src/provider/types.ts

```diff
@@ -120,3 +126,40 @@ export type ExecutionStats = {
   activeCursor: t.Slot
   finalCursor: t.Slot
 }
+
+export type SlotExecutionOutputFilter = {
+  status?: ExecutionOutputStatus
+  slotRange?: SlotRange
+  asyncPoolChangesFilter?: AsyncPoolChangesFilter
+  emptyExecutedDenounciationFilter?: boolean
+  eventFilter?: ExecutionEventFilter
+  executedOpsChangesFilter?: ExecutedOpsChangesFilter
+  ledgerChangesFilter?: LedgerChangesFilter
+}
+
+export type AsyncPoolChangesFilter = {
+  type?: AsyncPoolChangeType
+  handler?: string
+  destinationAddress?: string
+  emitterAddress?: string
+  canBeExecuted?: boolean
+  empty?: boolean
+}
+
+export type ExecutionEventFilter = {
+  callerAddress?: string
+  emitterAddress?: string
+  originalOperationId?: string
+  isFailure?: boolean
+  empty?: boolean
+}
+
+export type ExecutedOpsChangesFilter = {
+  empty?: boolean
+  operationId?: string
+}
+
+export type LedgerChangesFilter = {
+  empty?: boolean
+  address?: string
+}
```

Comment:
can you move theses types in a dedicated file? like create a `type.ts` or `filtersTypes.ts` in grpc folder

Reply:
this file was not in jsonRpc folders